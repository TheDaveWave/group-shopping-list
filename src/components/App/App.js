import React from 'react';
import axios from 'axios';
import {useState, useEffect} from 'react';
import Header from '../Header/Header.js'
import './App.css';
import ProductList from '../ProductList/ProductList.js';
import ProductForm from '../ProductForm/ProductForm.js';


function App() {
    const [productList, setProductList] = useState([]);

    // GET request to retrieve the products in our database.
    const getProducts = () => {
        console.log('In axios GET');
        axios.get('/products')
        .then(response => {
            console.log(response.data);
            setProductList(response.data);
        })
        .catch(err => {
            console.log('error in axios GET',err);
        });
    }

    console.log(productList);

    // call getProducts once on load.
    useEffect(() => {
        getProducts();
    }, []);


    return (
        <div className="App">
            <Header />
            <ProductForm getProducts={getProducts}/>
            <ProductList productList={productList} getProducts={getProducts}/>
        </div>
    );
}

export default App;
