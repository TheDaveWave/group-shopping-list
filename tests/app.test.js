import React from "react";
import { act as domAct } from "react-dom/test-utils";
import { create, act as testAct } from "react-test-renderer";
import App from "../src/components/App/App";
import { unmountComponentAtNode } from "react-dom";

describe("Renders App", () => {
  /* let container = null;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  }); */
  /* test("Snapple", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  }); */
  test("Run through app", () => {
    const fakeProducts = [
      {
        id: 1,
        name: "Balls",
        quantity: 2,
        unit: "balls",
        isPurchased: false,
      },
      {
        id: 2,
        name: "Sllab",
        quantity: 2,
        unit: "sllab",
        isPurchased: false,
      },
    ];

    /* jest.spyOn(global, "getProducts").mockImplementation(() => {
      Promise.resolve(() => {
        json: () => Promise.resolve(fakeProducts);
      });
    }); */

    let root;
    domAct(async () => {
      await testAct(() => {
        root = create(<App />);
      });
    });
    expect(root).toMatchSnapshot();
  });
});
