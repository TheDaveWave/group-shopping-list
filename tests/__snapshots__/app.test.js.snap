// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App Run through app 1`] = `
<div
  className="App"
>
  <header
    className="banner-header"
  >
    <div
      id="buyStuff"
    >
      <h1>
        BUY 
      </h1>
      <div
        id="tagHolder"
      >
        <div
          id="tag"
        >
          $
        </div>
      </div>
      <h1>
        STUFF
      </h1>
    </div>
  </header>
  <div
    id="productForm"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="nameIn"
      >
        Product Name:
      </label>
      <input
        id="nameIn"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="quantityIn"
      >
        Quantity:
      </label>
      <input
        id="quantityIn"
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
      <label
        htmlFor="unitIn"
      >
        Unit:
      </label>
      <input
        id="unitIn"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <button>
        Add Product
      </button>
    </form>
  </div>
  <div
    className="list-header"
  >
    <h2>
      Shopping List
    </h2>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
  <div
    className="item-container"
  />
</div>
`;
